lib/http2.c

```diff
@@ -1500,6 +1500,13 @@ static int on_begin_headers(nghttp2_session *session,
   return 0;
 }
 
+static void prepare_header_error_message(struct Curl_easy *data_s,
```

Comment:
Since the function does not really *prepare* the message, I would propose a small rename.

Also, we usually use plain `data` as name for the `struct Curl_easy` pointer a function works with.

```suggestion
static void report_header_error(struct Curl_easy *data,
```